.github/workflows/runtests.yml

```diff
@@ -151,6 +151,29 @@ jobs:
       # Test modules with specific dependencies
       - run: bin/test_optional_dependencies.py
 
+  # -------------------- Optional dependency tests for pypy ----------------- #
+
+  optional-dependencies-pypy:
+    needs: code-quality
```

Comment:
Can this be added using `matrix:` rather than as a separate listing?

The only difference with the job above is that pip doesn't install quite as many things but we could guard that with a conditional.

Reply:
I'll read the GitHub Actions doc to see how to achieve that.